oauthproxy.go

```diff
@@ -108,6 +109,8 @@ type OAuthProxy struct {
 	serveMux          *mux.Router
 	redirectValidator redirect.Validator
 	appDirector       redirect.AppDirector
+
+	shouldBase64Encode bool
```

Comment:
Maybe something like shouldEncodeState is clearer? @tuunit any opinions?

Reply:
Agree, the name is not helpful. Fixed